d20/Manual/Shell.py

```diff
@@ -435,191 +457,206 @@ def run(self):
-        self.item = item
-
-    def write_list(self, out):
-        sys.stdout.write("\n%s %d:\n" % (self._type.capitalize(),
-                                         self.item.id))
-        sys.stdout.write("--------------\n")
-        sys.stdout.write(out)
-        sys.stdout.write("--------------\n\n")
+        if item.id is not None:
+            # Append item to depth list
+            self._type: str = _type
+            self.depthList.append(item)
+            self.prompt: str = "%s %d > " % (self._type, item.id)
+            self.item: Fact = item
+        else:
+            sys.stdout.write("Something went wrong, Fact had no ID\n")
+
+    def write_list(self, out: str) -> None:
+        if self.item.id is not None:
+            sys.stdout.write("\n%s %d:\n" % (self._type.capitalize(),
+                                             self.item.id))
+            sys.stdout.write("--------------\n")
+            sys.stdout.write(out)
+            sys.stdout.write("--------------\n\n")
+        else:
+            sys.stdout.write("Something went wrong, Fact had no ID\n")
 
-    def _find_info(self):
-        created = tsTodt(self.item.created)
-        item_info = collections.OrderedDict([
+    def _find_info(self) -> OrderedDict:
+        created: str = tsTodt(self.item.created)
+        item_info: OrderedDict = collections.OrderedDict([
                         ('id', self.item.id),
                         ('type', self.item._type),
                         ('creator', self.item.creator),
                         ('created', created)])
 
-        for fieldName in self.item._fields_:
-            descriptor = getattr(self.item, '_%s__' % fieldName)
-            if (isinstance(descriptor, SimpleField) or
-                    isinstance(descriptor, NumericalField) or
-                    isinstance(descriptor, StrOrBytesField)):
-                try:
-                    item_info[fieldName] = descriptor.getShell()
-                except AttributeError:
-                    item_info[fieldName] = None
-            else:
-                if 'fields' not in item_info:
-                    item_info['fields'] = list()
-                item_info['fields'].append(fieldName)
+        if self.item._fields_:
+            for fieldName in self.item._fields_:
+                descriptor = getattr(self.item, '_%s__' % fieldName)
+                if (isinstance(descriptor, SimpleField) or
+                        isinstance(descriptor, NumericalField) or
+                        isinstance(descriptor, StrOrBytesField)):
+                    try:
+                        item_info[fieldName] = descriptor.getShell()
+                    except AttributeError:
+                        item_info[fieldName] = None
+                else:
+                    if 'fields' not in item_info:
+                        item_info['fields'] = list()
+                    item_info['fields'].append(fieldName)
 
-        return item_info
+            return item_info
```

Comment:
is this the correct indentation?  If the conditional evaluates false above the function will return `None` and not and `OrderedDict` so the type hint will be wrong or should it be an `Optional`

Reply:
I think I have the function raise an exception if the conditional evaluates false, so I don't think an `Optional` is needed?

Reply:
Ah, I see, I missed that with the `do_info` move the diff did below, seems good then as it would be a problem if the fields were not present.